stan/math/rev/core/arena_matrix.hpp

```diff
@@ -73,6 +73,50 @@ class arena_matrix : public Eigen::Map<MatrixType> {
     *this = other;
   }
 
+  /**
+   * Constructs `arena_matrix` from an expression, then send it to either the
+   * object stack or memory arena.
+   * @tparam T A type that inherits from Eigen::DenseBase that is not an
+   * `arena_matrix`.
+   * @param other expression
+   * @note When T is both an rvalue and a plain type, the expression is moved to
+   * the object stack. However when T is an lvalue, or an rvalue that is not a
+   * plain type, the expression is copied to the memory arena.
+   */
+  template <typename T, require_eigen_t<T>* = nullptr,
+            require_not_arena_matrix_t<T>* = nullptr>
+  arena_matrix(T&& other)  // NOLINT
```

Comment:
Would it be possible to split this to two constructors using `require_t<std::is_rvalue_reference<T>::value>` rather than having the constructor instantiate and call a lambda? Just feels a little unnecessarily complex

Reply:
Yes that's much cleaner. Though I still kept the immediately evaluated lambda as that keeps the constructor empty which I find kind of nice